pkg/pool-hooks/contracts/StableSurgeHookExample.sol

```diff
@@ -0,0 +1,235 @@
+        uint256 invariant = StableMath.computeInvariant(amp, params.balancesScaled18);
+        uint256 weightAfterSwap;
+        if (params.kind == SwapKind.EXACT_IN) {
+            uint256 amountCalculatedScaled18 = StableMath.computeOutGivenExactIn(
+                amp,
+                params.balancesScaled18,
+                params.indexIn,
+                params.indexOut,
+                params.amountGivenScaled18,
+                invariant
+            );
+            // Swap fee is always a percentage of the amountCalculated. On ExactIn, subtract it from the calculated
+            // amountOut. Round up to avoid losses during precision loss.
+            uint256 swapFeeAmountScaled18 = amountCalculatedScaled18.mulUp(staticSwapFeePercentage);
+            amountCalculatedScaled18 -= swapFeeAmountScaled18;
+            weightAfterSwap = getWeightAfterSwap(
+                params.balancesScaled18,
+                params.indexIn,
+                params.amountGivenScaled18,
+                amountCalculatedScaled18
+            );
+        } else {
+            uint256 amountCalculatedScaled18 = StableMath.computeInGivenExactOut(
+                amp,
+                params.balancesScaled18,
+                params.indexIn,
+                params.indexOut,
+                params.amountGivenScaled18,
+                invariant
+            );
+            // To ensure symmetry with EXACT_IN, the swap fee used by ExactOut is
+            // `amountCalculated * fee% / (100% - fee%)`. Add it to the calculated amountIn. Round up to avoid losses
+            // during precision loss.
+            uint256 swapFeeAmountScaled18 = amountCalculatedScaled18.mulDivUp(
+                staticSwapFeePercentage,
+                staticSwapFeePercentage.complement()
+            );
+
+            amountCalculatedScaled18 += swapFeeAmountScaled18;
+            weightAfterSwap = getWeightAfterSwap(
+                params.balancesScaled18,
+                params.indexIn,
+                amountCalculatedScaled18,
+                params.amountGivenScaled18
+            );
+        }
+
+        uint256 thresholdBoundary = getThresholdBoundary(params.balancesScaled18.length, _threshold);
+        if (weightAfterSwap > thresholdBoundary) {
+            return (true, getSurgeFee(weightAfterSwap, thresholdBoundary, staticSwapFeePercentage, _surgeCoefficient));
+        } else {
+            return (true, staticSwapFeePercentage);
+        }
+    }
+
+    /**
+     * Defines the range in which surging will not occur.
+     * @dev An expected value for threshold in a 2 token (n=2) would be 0.1.
+     * This would mean surging would occur if any token reaches 60% of the total of balances.
+     * @param numberOfAssets Number of assets in the pool.
+     * @param threshold Theshold value.
+     */
+    function getThresholdBoundary(uint256 numberOfAssets, uint256 threshold) public pure returns (uint256) {
+        return 1e18 / numberOfAssets + threshold;
```

Comment:
```suggestion
        return FixedPoint.ONE / numberOfAssets + threshold;
```
We've been using the defined constant for this vs. hard-coding numbers. So this would create an upper bound on the threshold then. With 2 tokens, the max threshold would be 0.5? With 8 tokens, 0.875? Assuming the boundary doesn't go over 1?

How can this be set on the hook then? If I understand it correctly, and the number of pool tokens constrains the maximum threshold, it would be different for every pool then - or at least every pool size. So, you'd set the threshold for all 2-token pools, then all 3-token pools, etc. (I think I don't understand this well.)